TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -58,39 +62,130 @@ public class CarSpawner : MonoBehaviour
         private List<GameObject> _vehicleTypes;
 
         private List<Lane> _lanes = new List<Lane>(); 
-        private List<float> _lengths = new List<float>();
-        private List<float> _ratios = new List<float>();
         private List<int> _indexes = new List<int>();
-        private List<int> _maxCarsPerLane = new List<int>();
 
         private LaneNode _laneNodeCurrent;
 
         private GameObject _currentCar;
 
         private int _carCounter = 0;
-        private float _offset = 0;
         private float _carLength;
 
         private bool _spawned = false;
-        
-        private void Start()
+
+        private List<CarsForLane> spawnableLanes = new List<CarsForLane>();
+
+        // Create a empty gameobject to hold all cars
+        private GameObject _carContainer;
+
+        public struct SpawnableLaneNode
+        {
+            private float _distance;
+            private LaneNode _node;
+
+            public float Distance { get => _distance; set => _distance = value; }
+            public LaneNode Node { get => _node; set => _node = value; }
```

Comment:
No reason for this when the getters and setters are public and contain no additional logic